src/components/common/PeopleAndRoles/PeopleAndRoles.vue

```diff
@@ -605,15 +654,20 @@ export default class PeopleAndRoles extends Mixins(CommonMixin, DateMixin) {
     // (we update this record right in the temp list)
     const person = tempList[index]
 
-    // just set the action (ie, soft-delete)
-    // person will be filtered out on file and pay
-    person.actions = [ActionTypes.REMOVED]
+    if (person.actions?.includes(ActionTypes.ADDED)) {
+      // splice out the person
+      tempList.splice(index, 1)
```

Comment:
Bugfix: In some cases, if you added a person and then removed them, they should still show in the list as Removed (and Undo wouldn't work).

Example with previous code:

![image](https://user-images.githubusercontent.com/10002889/172733559-f92ec26e-6a61-4084-bffe-577cffc2aa06.png)